dev-tools/mage/crossbuild.go

```diff
@@ -63,6 +63,13 @@ func WithTarget(target string) func(params *crossBuildParams) {
 	}
 }
 
+// InDir specifies the base directory to use when cross-building.
```

Comment:
I think it's important to document that the directory is automatically suffixed with the current sub-directory path relative to the repo root. I wasn't expecting this. 

I was expecting to the see it used like `InDir(filepath.Join("x-pack", BeatName))`. Or make the signature be `InDir(path ...string)` and automatically invoke `filepath.Join(path...)` in the implementation.

Reply:
I can do that, but I don't have the `BeatName` in `CrossBuild`, it's just using the current subidr. In `filebeat/magefile.go` I could pass `InDir("filebeat")` for oss build and `InDir("x-pack", "filebeat")` in the xpack. WDYT?

Reply:
Got it, I updated this again, I think I finally understood what you meant :)

Reply:
LGTM